PR HW-10 From Branch_004 to Main

AleksanderQA opened the pull request.

AleksanderQA committed:
Created: Commit-HW-09
Api automated unit tests
Test Case according to Delete method
Added dependencies: Rest Assured, Params, API, maven plugin

AleksanderQA committed:
Commit HW-10.
Added:
Parametrized API unit test
Extended Test case
Annotations: @CsvSource
@Tag used for "HW-10"

vladimir-tl commented on src/test/java/Api/RestApiMocked.java:
```diff
@@ -0,0 +1,152 @@
+    public void requestGetWithUsernameAndPasswordToReceiveApiKey(String username, String password) {
+        String responseString =
+                given()
+                        .queryParam("username", username)
```
классная находка - попрошу поделиться на уроке со всеми

AleksanderQA committed:
Commit HW-10.
Added:
Parametrized API unit test
Extended Test case
Annotations: @CsvSource
@Tag used for "HW-10"
---------------------
Changed HttpStatus fields and added clarifications
for method names
